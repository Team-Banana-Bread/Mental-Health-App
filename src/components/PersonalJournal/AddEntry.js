import React, {useState} from 'react';
import {View, Text, StyleSheet, TextInput, TouchableOpacity} from 'react-native';
import { Button } from '@ui-kitten/components';
import Icon from 'react-native-vector-icons/dist/FontAwesome';

const AddEntry = ({title, addItem}) => {
    const [text, setText] = useState('');
    const onChange = (textValue) => setText(textValue);
    return(
        <View>
            <TextInput placeholder="New Entry" style={styles.input} 
                        onChangeText={onChange} blurOnSubmit={false} multiline={true} returnKeyType='none'/>
            <Button style={styles.btn} onPress={() => addItem(text)}>
                <Text style={styles.btnText}><Icon name="plus" size={20}/>Add Entry</Text>
            </Button>
        </View>
    );
};

const styles = StyleSheet.create({
    input:{
        height: 60,
        padding: 8,
        fontSize: 16,
    },
    btn: {
        backgroundColor: '#c2bad8',
        padding: 9,
        margin: 5,
        borderColor: 'transparent'
    },
    btnText: {
        color: 'darkslateblue',
        fontSize: 20,
        textAlign: 'center',
    },
});

export default AddEntry;