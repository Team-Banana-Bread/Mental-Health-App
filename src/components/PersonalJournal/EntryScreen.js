import React from 'react';
import {
    StyleSheet,
    Text,
    View,
    TextInput,
    ScrollView,
    TouchableOpacity,
} from 'react-native';

export default class EntryScreen extends React.Component {
    static navigationOptions = {
        title: 'New Entry',
        headerShown: false,
    };

    state = {
        entryText: '',
    }

    render() {
        var d = new Date();
        return (
            <View style={styles.container}>
                <View style={styles.header}>
                    <Text style={styles.headerText}></Text>
                    <TouchableOpacity
                        onPress={() => {
                            this.props.navigation.goBack()
                        }}
                        style={styles.backButton}>
                        <Text style={styles.backButtonText}> Back </Text>
                    </TouchableOpacity>
                </View>
                <ScrollView>
                    <TextInput
                        style={styles.textInput}
                        onChangeText={(entryText) => this.setState({ entryText })}
                        value={this.state.entryText}
                        placeholder='New Entry'
                        multiline={true}
                        numberOfLines={30}
                        placeholderTextColor='grey'
                        underlineColorAndroid='transparent'>
                    </TextInput>
                </ScrollView>
                <View style={styles.buttonSection}>
                    <TouchableOpacity
                        onPress={() => {
                            receivedValue(entryArray.push({
                                'date': d.getFullYear() +
                                    "/" + (d.getMonth() + 1) +
                                    "/" + d.getDate(),
                                'entry': this.state.entryText
                            }))
                            this.props.navigation.goBack()
                        }}
                        style={styles.addButton}>
                        <Text style={styles.addButtonText}>Enter</Text>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    headerText: {
        color: 'white',
        fontSize: 18,
        padding: 26,
    },
    scrollContainer: {
        flex: 1,
        marginBottom: 100,
    },
    textInput: {
        alignSelf: 'stretch',
        textAlignVertical: "top",
        color: 'black',
        padding: 20,
        backgroundColor: 'white',
        borderTopWidth: 2,
        borderTopColor: '#ededed',
    },
    addButton: {
        position: 'absolute',
        zIndex: 11,
        flexDirection: 'column',
        bottom: 20,
        right: 20,
        backgroundColor: '#97D3BF',
        padding: 10,
        borderRadius: 5,
        elevation: 8,
    },
    addButtonText: {
        color: '#fff',
        fontSize: 24,
    },
    header: {
        backgroundColor: '#97D3BF',
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        borderBottomWidth: 5,
        borderBottomColor: '#AECCC2',
    },
    backButton: {
        position: 'absolute',
        zIndex: 11,
        flexDirection: 'row',
        bottom: 10,
        left: 10,
        backgroundColor: '#97D3BF',
        padding: 10,
        borderRadius: 5,
        elevation: 8,
    },
    backButtonText: {
        color: '#fff',
        fontSize: 17,
    },
});