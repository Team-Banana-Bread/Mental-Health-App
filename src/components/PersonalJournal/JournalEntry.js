import React from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity
} from 'react-native';

export default class JournalEntry extends React.Component {
    render() {
        return (
            <View key={this.props.keyval} style={styles.entry}>

                <Text styles={styles.entryText}>{this.props.val.date}</Text>
                <Text styles={styles.entryText}>{this.props.val.entry}</Text>

                <TouchableOpacity onPress={this.props.deleteMethod} style={styles.entryDelete}>
                    <Text style={styles.entryDeleteText}>Delete</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    entry: {
        position: 'relative',
        padding: 20,
        paddingRight: 100,
        borderBottomWidth: 2,
        borderBottomColor: '#ededed',
    },
    entryText: {
        paddingLeft: 20,
        borderLeftWidth: 10,
        borderLeftColor: '#E91E63',
    },
    entryDelete: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#2980b9',
        padding: 10,
        top: 10,
        bottom: 10,
        right: 10,
    },
    entryDeleteText: {
        color: 'white',
    }
});
