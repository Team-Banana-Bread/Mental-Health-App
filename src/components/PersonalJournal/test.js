import React from 'react';
import renderer from 'react-test-renderer';

import PersonalJournal from './PersonalJournal';

test('renders correctly', () => {
    const tree = renderer.create(<PersonalJournal />).toJSON();
    expect(tree).toMatchSnapshot();
  });