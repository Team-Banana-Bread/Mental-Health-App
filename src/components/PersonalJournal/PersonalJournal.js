import React, {useState} from 'react';
import {
    StyleSheet,
    Text,
    View,
    TextInput,
    ScrollView,
    TouchableOpacity,
    Alert
} from 'react-native';
import { Button } from '@ui-kitten/components';
import JournalEntry from './JournalEntry';
import AddEntry from './AddEntry';
import { FlatList } from 'react-native-gesture-handler';
import { v4 as uuidv4 } from 'uuid';    

const PersonalJournal = () => {
    const [items, setItems] = useState([
    ]);

    const deleteItem = (id) => {
        setItems(previousItems => {
            return previousItems.filter(item => item.id != id);
        });
    };

    const addItem = (text) => {
        if(!text){
            Alert.alert('Error', 'Please enter something', [{text: "OK", onPress: () => console.log("OK Pressed")}], {cancelable: true }
            )}else{
            setItems(previousItems => {
                return [{id: uuidv4(), text},...previousItems];
            });
        }
    }

        return(
            <View style = {styles.container}>
                    <Text style={styles.header}> JOURNAL </Text>
                    <AddEntry addItem={addItem}/>
                    <FlatList
                        data={items} renderItem={({item}) => 
                        <JournalEntry item={item}
                        deleteItem={deleteItem}/> 
        }/>
            </View>
        );
        
};


const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    header: {
        backgroundColor: 'lightsteelblue',
        alignItems: 'center',
        justifyContent: 'center',
        borderBottomWidth: 5,
        borderBottomColor: 'lightsteelblue',
        color: 'white',
        fontSize: 20,
        padding: 26,
        textAlign: 'center'
    },
    scrollContainer: {
        flex: 1,
        marginBottom: 100,
    },
    addButton: {
        position: 'absolute',
        zIndex: 11,
        right: 20,
        bottom: 20,
        backgroundColor: 'lightsteelblue',
        width: 90,
        height: 90,
        borderRadius: 50,
        alignItems: 'center',
        justifyContent: 'center',
        elevation: 8,
        borderColor: "transparent"
    },
    addButtonText: {
        color: '#fff',
        fontSize: 24,
    },
    entryText: {
        paddingLeft: 20,
        borderLeftWidth: 10,
        borderLeftColor: '#E91E63',
        borderBottomWidth: 2,
        borderBottomColor: '#ededed',
    },
});

export default PersonalJournal;