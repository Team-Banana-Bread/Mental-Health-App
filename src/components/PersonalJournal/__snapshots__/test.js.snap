// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#97D3BF",
        "borderBottomColor": "#AECCC2",
        "borderBottomWidth": 5,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "padding": 26,
        }
      }
    >
       JOURNAL 
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#97D3BF",
        "borderRadius": 50,
        "bottom": 20,
        "elevation": 8,
        "height": 90,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 20,
        "width": 90,
        "zIndex": 11,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 24,
        }
      }
    >
      +
    </Text>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "marginBottom": 100,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
